index.js

```diff
@@ -1,4 +1,16 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import EresourceSearch from './src/EresourceSearch';
+import EresourceSearchModal from './src/Modal';
 
-export default props => <EresourceSearch {...props} />;
+function PluginFindEresource(props) {
+  return (props.justDisplayModal ?
+    <EresourceSearchModal {...props} /> :
```

Comment:
@MikeTaylor When using the justDisplayModal prop in the current approach, I think the user won’t be able to close the modal on clicking the close button? The onClose logic currently lies in a lower level component and it’s not passed in via a prop to the Modal?

Reply:
Thanks for looking at this!

No, all the props are passed through as `{...props}`, so when I invoke this I pass in my own `onClose` function. Here's how it looks in my code (slighty simplified for ease of reading):
```
        const [recordToEdit, setRecordToEdit] = useState();
        [...]
        return (
          {recordToEdit &&
            <Pluggable
              type="find-eresource"
              justDisplayModal
              open
              onClose={() => setRecordToEdit(undefined)}
              onEresourceSelected={someFunction}
            >
              Find eResource
            </Pluggable>
          }
```
(And elsewhere, some code calls `setRecordToEdit(rec)` to open the modal.)

Reply:
To be clearer: if you invoke the plugin in `justDisplayModal` form, then you are undertaking to manage the whole of the open/close logic yourself, rather than having the plugin do it for you.

Reply:
I see. Although, I am still not convinced with the API enhancement as to how a new comsumer of this plugin would possibly know the props he'd have to pass down with this full manual override approach? In your case, `onClose` is passed down and that satisifies this specific use case, but that specific prop is something you figured as you have looked at the implemenation details (Its not a part of the API as it stands now). So, a new consumer has to look at the implementation details right? Am I making sense?

Reply:
Ah, you're right — more documentation is needed to make this clear. I'll add to the README.

Reply:
Try this: https://github.com/folio-org/ui-plugin-find-eresource/blob/add-justDisplayModal-prop/README.md#controlled-invocation

Reply:
Hmm, but this would still need additional props as a part of the readme if we are taking this approach? namely: `open`, `modalRef`, etc. So, question is do you want to make this a controlled plugin? i.e. have the `open`, `close` state managable from the outside? or can we have this remain `uncontrolled` but you just need a way to not render the trigger button but just default the modal to an `open` state?

Reply:
No, none of those additional props are necessary. I know this not merely by abstract reasoning but because _I have functioning code that uses this plugin by controlled invocation_. See https://github.com/folio-org/ui-plugin-eusage-reports/blob/3d51b5b19ac897dece291b8992fdc44776116ea0/src/components/MatchEditor.js#L160-L169

I wish you would trust me to have thought this through and done what I need. For example, I tried merely starting it in the open state — that's no good because when you close it, you're left with the visible trigger.

Reply:
Yeah, I mean I am sure its working as expected. I am just trying to figure an easier way (if any). How about adding a new prop to not render the trigger at all? 
eg: [here](https://github.com/folio-org/ui-plugin-find-eresource/blob/master/src/EresourceSearch.js#L55) we could do, 
`if (this.props.renderWithoutTrigger) return null` or similar. Would that serve your purpose? along with a `defaultOpen` prop that you initially set in the constructor to manage the `open` state?

Reply:
Actually, for the above approach, you could just pass a `renderTrigger` function that just returns `null` instead of the additional `renderWithoutTrigger ` prop

Reply:
No, because the built-in `onClose` function will not make the application-level state-change I need.

Reply:
Okay, then I guess we just need an `onClose` callback passed in as a prop? that if present is called [here](https://github.com/folio-org/ui-plugin-find-eresource/blob/master/src/EresourceSearch.js#L35).
`this.props.onClose && this.props.onClose()`

Reply:
I don't understand why you are so keen to replace my simple, proven-to-work change, which touches _one file_, with something different that has not been used and which will require my calling code to bend over backwards with a weird hybrid calling mechanism.

Reply:
What you propose here would involve two separate mechanisms for determining whether the overlay is open or not, with consequence confusion between them. It is much simpler just to access the model directly, with a single layer of control from outside this plugin. Which is why I did it that way.

Reply:
So what I am proposing is more an uncontrolled way of maintaining that plugin, which is how it is right now and I believe would suffice your requirement? Now, once the control is out of the plugin, its an unncessary overhead at the app level dont you think? (Like you need to maintain the local state for open/close, etc). Thats just additional inversion of control when its not necessary in this case in my opinion.

From what I understand, all you need is just a`renderTrigger` prop that returns null, a `defaultOpen` and an `onClose` prop, the latter 2 which provide no harm in adding and in fact a betterment of the already existing API? and you then wont also need the conditional in the `index.js` file.  Thoughts?